### PR TITLE
Implement igcollect plugin for puppetserver

### PR DESCRIPTION
**Pull request by Woellchen**

### CODE REVIEW

File: igcollect/puppetserver.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    for metric in jruby:
+        print_single(template, 'master.jruby-metrics', metric, jruby[metric])
+
+    # Skip profiler metrics if not requested
+    if not args.with_profiler:
+        return
+
+    # Collect profiler metrics
+    profiler = data['puppet-profiler']['status']['experimental']
+    descs = [
```
Comment (TheFRedFox) [inline igcollect/puppetserver.py]:
``` python
descs = {
    'function-metrics': ['function', 'mean'],
    ...
}
```

File: igcollect/puppetserver.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    """CLI"""
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='puppetserver')
+    parser.add_argument(
+        '--url',
+        default='https://localhost:8140/status/v1/services?level=debug'
+    )
+    parser.add_argument('-p', '--with-profiler', action='store_true')
+    parser.add_argument('-n', '--no-ssl-verify', action='store_true')
+    parser.add_argument('-c', '--cert')
```
[the remark this answers is not in the document]
Comment (Woellchen) [reply igcollect/puppetserver.py]: thanks! i just saw that i cannot pass streams to the ssl context, so i would leave file validation and loading to the module and rather not deal with it

File: igcollect/puppetserver.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+import re
+
+
+def parse_args():
+    """CLI"""
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='puppetserver')
+    parser.add_argument(
+        '--url',
+        default='https://localhost:8140/status/v1/services?level=debug'
```
[the remark this answers is not in the document]
Comment (Woellchen) [reply igcollect/puppetserver.py]: yes it is

[the remark this answers is not in the document]
Comment (Woellchen) [reply igcollect/puppetserver.py]: yup

File: igcollect/puppetserver.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        {'resource-metrics': ['resource', 'mean']},
+        {'catalog-metrics': ['metric', 'mean']},
+        {'inline-metrics': ['metric', 'count']},
+    ]
+    print_multiple(template, descs, profiler)
+
+
+def print_multiple(template, descs, data):
+    """Print a metrics tree from the API"""
+    for desc in descs:
```
[the remark this answers is not in the document]
Comment (Woellchen) [reply igcollect/puppetserver.py]: no, of course not. changed

**Pull request merged**